packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -181,6 +181,22 @@ def get_ps_output():
         assert len(get_ps_output().split()) <= 4, 'UCR app has more than 4 processes running in its pid namespace'
 
 
+def test_memory_profiling(dcos_api_session):
+    # Test that we can fetch raw memory profiles
+    master_ip = dcos_api_session.masters[0]
+    r0 = dcos_api_session.get(
+        '/memory-profiler/start', host=master_ip, port=5050)
+    assert r0.status_code == 200
+
+    r1 = dcos_api_session.get(
+        '/memory-profiler/stop', host=master_ip, port=5050)
+    assert r1.status_code == 200
+
+    r2 = dcos_api_session.get(
+        '/memory-profiler/download/raw', host=master_ip, port=5050)
+    assert r2.status_code == 200
```

Comment:
We could check that `r2.text` is not empty or some other indication that the response was as intended.

Reply:
Let's not assume any specific format for the heap profiles. Since i'm not leaving any time for measurements in order to avoid sleeping, an empty response doesn't necessarily indicate failure.